geode-deployment/geode-deployment-legacy/src/integrationTest/java/org/apache/geode/internal/deployment/JarDeployerIntegrationTest.java

```diff
@@ -202,6 +208,26 @@ public void undeploy() throws Exception {
         .isInstanceOf(ClassNotFoundException.class);
   }
 
+  @Test
+  public void deploy2JarsSetCurrentClassloaderTo1stAndLoadClassFrom2nd() throws Exception {
+
+    // deploy def-1.0.jar
+    // deploy base.jar
+    jarDeployer.deploy(semanticJarVersion1);
+    DeployedJar deployedJar = jarDeployer.deploy(baseJar);
+
+    assertThat(deployedJar).isNotNull();
+
+    ClassPathLoader oldLoader = ClassPathLoader.getLatest();
+
+    ClassLoader cl = oldLoader.getClassloaderForArtifact("def"); // set current classloader to 1st
+
+    cl.loadClass("jddunit.function2.ExceptionB"); // load extended class and base class
+    cl.loadClass("jddunit.function1.ExceptionA"); // load base class
```

Comment:
I think it is best to have the comments before the code statements. Otherwise, if the line of code grows, the comment may be split in several lines.